connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -202,6 +198,8 @@ public void removeMetrics() {
     public void cancel() {
         super.cancel();
         offsetReader.close();
+        // Run on a separate thread to avoid potentially blocking the herder thread
+        new Thread(() -> closeProducer(0)).start();
```

Comment:
nit: i don't want to have to guess what 0 means
```suggestion
        new Thread(() -> closeProducer(Duration.ofSeconds(0))).start();
```
also, don't we have an executor for this sort of stuff...

Reply:
Yeah, we can probably use an executor for this. I was on the fence about the necessity of doing this in a separate thread at all (hence the half-assing here) but then I looked deeper into the `KafkaProducer::close` logic and saw that that included closing its interceptors, key serializer, value serializer, etc. Any time user code is called we should assume it can block forever (probably worth adding as a comment here), so yeah, since I'm more confident in the necessity of asynchronously closing the producer I'll expand this to use an executor.